packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts

```diff
@@ -180,19 +180,19 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
     const cacheResult = cacheRetriever
       ? await this.fetchDataSpaceAnalysisFromCache(cacheRetriever, actionState)
       : undefined;
-    const engineClient = this.graphManager.engine.getEngineServerClient();
+    const engine = this.graphManager.engine;
```

Comment:
This is an example of leaky abstraction. Maybe there's something we need to adjust to expose method from engine to do this operation rather than directly calling a client. For example, let's say the engine can be abstracted further to have capabilities pluggable by extensions (i.e. `PureProtocolProcessorPlugin`), then we can "cast" like below

```ts
const engine = V1_DSL_DataSpace_getEngine(this.graphManager.engine);
```

then after this point, engine is aware of capabilities from `DSL_DataSpace`.

---

Having said that, I think that's overkill for this PR, but something to think about to enhance further later.

Reply:
Yeah I agree. There's several instances of plugins and some other classes directly using the engine server client, which I think is not the best way to handle things. Ideally I think things should only be interacting with the engine and not interacting with implementation details. For now, perhaps we can allow the plugins to get the engine server client and check that the type is of `V1_EngineServerClient` so it can access the methods needed without adding things to the abstract engine interface.

Reply:
yeah I think for now, it's okay to expose client, but let's keep this to minimum, hence, all the `getBaseUrl...` methods can be scraped